optimum/habana/transformers/models/falcon/modeling_falcon.py

```diff
@@ -373,8 +649,17 @@ class GaudiFalconModel(FalconModel):
     - set past_key_values_length=0 when token_idx is used (with static input shape)
     - add new arg tgt_len to _expand_mask because past_key_values_length is no longer valid with token_idx
     - use old version of _make_causal_mask to workaround toch.triu that is not supported in Synapse
```

Comment:
wondering, triu isnt supported but tril is? because tril is used in ScaledDotProductAttention
Or maybe this comment is old

Reply:
you are right, the usage of '_make_causal_mask' is removed when transformers was upgraded to 4.27.0. i will remove old comments